src/auto-evo/steps/FindBestMutation.cs

```diff
@@ -17,12 +17,16 @@ public class FindBestMutation : VariantTryingStep
 
         private readonly Mutations mutations = new();
 
-        public FindBestMutation(AutoEvoConfiguration configuration, PatchMap map, Species species,
+        private WorldGenerationSettings settings;
+
+        public FindBestMutation(AutoEvoConfiguration configuration,
+            WorldGenerationSettings settings, PatchMap map, Species species,
```

Comment:
I think here the use of WorldGenerationSettings is pretty unclear why it is needed here. I think the design would be clearer if you create some interface to store the difficulty data or the world configuration that is needed here. Something like IWorldConfiguration or IDifficultyOption or something like that, or maybe even passing in the current game here?

Reply:
Passing in the current game sounds simplest to me.